rs/xnet/payload_builder/src/lib.rs

```diff
@@ -879,11 +883,24 @@ impl XNetPayloadBuilderImpl {
 ///
 /// In order to prevent mutual stalling, only applies to incoming NNS
 /// streams; and to `Application`-subnet-to-`System`-subnet streams.
-pub fn get_msg_limit(subnet_id: SubnetId, state: &ReplicatedState) -> Option<usize> {
+pub fn get_msg_limit(
+    subnet_id: SubnetId,
+    state: &ReplicatedState,
+    expected: Option<&ExpectedIndices>,
+    slice_begin: StreamIndex,
+) -> Option<usize> {
+    let get_limit = || -> Option<usize> {
+        const MAX_SIGNALS_STREAM_COUNT: usize = 50_000;
```

Comment:
This should be more than a constant hidden down in some function. If nothing else, it should be a documented constant somewhere at the top of the crate / module.

Also `MAX_SIGNALS_STREAM_COUNT` is hard to read. `MAX_SIGNAL_COUNT`? `MAX_STREAM_SIGNALS`? `MAX_STREAM_SIGNAL_COUNT`? (Although the latter is kind of redundant).